lib/containers/runtime.pm

```diff
@@ -32,27 +34,73 @@ sub _rt_script_output {
     return script_output($self->runtime . " " . $cmd, @args);
 }
 
-=head2 build
+=head2 create_container($image, $name, [%args])
 
-Build a container.
+Creates a container.
+C<image> the name of the image we create the container from.
+C<name> is the given name of the created container.
+C<args> can be anything you want to run in the container, similar to C<run>. for 
+instance: C<docker create -it tumbleweed bash>
+
+=cut
+sub create_container {
+    my ($self, $image, $name, $args) = @_;
+    die 'wrong number of arguments' if @_ < 3;
+    $self->_rt_assert_script_run("container create --name $name $image $args", 300);
+    record_info "$name container created", "";
+}
+
+=head2 start_container($image_name)
+
+Starts container named C<image_name>.
+
+=cut
+sub start_container {
+    my ($self, $image_name) = @_;
+    die 'wrong number of arguments' if @_ < 2;
+    $self->_rt_assert_script_run("container start $image_name");
+    record_info "$image_name container started", "";
+}
+
+=head2 halt_container($container_name)
+
+Blocks a container until exits.
+C<container_name> which runs.
+https://docs.docker.com/engine/reference/commandline/wait/
+
+=cut
+sub halt_container {
+    my ($self, $container_name) = @_;
+    die 'wrong number of arguments' if @_ < 2;
+    $self->_rt_assert_script_run("wait $container_name");
+    record_info "$container_name container is blocked.", "";
+}
+
+
+=head2 build($dockerfile_path, $container_tag, [%args])
+
+Build a container from a Dockerfil.
 C<dockerfile_path> is the directory with the Dockerfile as well as the root of the build context.
 C<container_tag> will be the name of the container.
+Give C<timeout> if you want to change the timeout passed to C<assert_script_run>. Default for containers is 300.
 
 =cut
 sub build {
-    my ($self, $dockerfile_path, $container_tag) = @_;
+    my ($self, $dockerfile_path, $container_tag, %args) = @_;
     die 'wrong number of arguments' if @_ < 3;
+
     #TODO add build with URL https://docs.docker.com/engine/reference/commandline/build/
-    $self->_rt_assert_script_run("build -f $dockerfile_path/Dockerfile -t $container_tag $dockerfile_path", 300);
+    $self->_rt_assert_script_run("build -f $dockerfile_path/Dockerfile -t $container_tag $dockerfile_path", $args{timeout} // 300);
     record_info "$container_tag created", "";
 }
 
-=head2 up
+=head2 up($image_name, [mode, name, remote, keep_container, timeout])
```

Comment:
I know you want to keep the same names, but I think we can change the name of this function as just `up` is not very descriptive. I would use `run_container` or `run_command`. I'm ok to do it in a different PR.

Reply:
i agree. what about `launch_container`? otherwise `run_container` is good.

Reply:
I strongly suggest `run` or `run_container` as we do `$runtime run` so it should match otherwise we will be all lost.